speakleash_dashboard.py

```diff
@@ -1,238 +1,350 @@
+            margin-right: auto;
+        }
+    </style>
+    """, unsafe_allow_html=True)
+
+    st.markdown('<div class="center-text"><h3>An open collaboration project to build a data set for Large Language Modeling \
+                with a volume of at least 1TB comprised of diverse texts in Polish. Our aim is to enable machine learning \
+                research in our beautiful, native language and to train a Generative Pre-trained Transformer Model from \
+                collected data.</h3></div>', unsafe_allow_html=True)
+
+# Summary table + progress plot 
+with row3:
+    st.markdown('<div class="center-text"><h2>So far we managed to collect:</h2></div>', 
+                unsafe_allow_html=True)
+
+    # Summary table
+    table_data = {
+        'Total documents': ["{:,}".format(total_documents).replace(",", " ")],
+        'Total characters': ["{:,}".format(total_characters).replace(",", " ")],
+        'Total sentences': ["{:,}".format(total_sentences).replace(",", " ")],
+        'Total words': ["{:,}".format(total_words).replace(",", " ")],
+        'Total verbs': ["{:,}".format(total_verbs).replace(",", " ")],
+        'Total nouns': ["{:,}".format(total_nouns).replace(",", " ")],
+        'Total punctuations': ["{:,}".format(total_punctuations).replace(",", " ")],
+        'Total symbols': ["{:,}".format(total_symbols).replace(",", " ")],
+        'Total stopwords': ["{:,}".format(total_stopwords).replace(",", " ")]
+    }
+  
+    df2 = pd.DataFrame(table_data)
+   
+    st.markdown('<div style="display: flex; justify-content: center;">' + \
+                df2.to_html(col_space="auto",
+                            justify='center',
+                            classes='table-responsive', 
+                            index=False).replace('<table', '<table style="white-space: nowrap; text-align: center;"') + \
+                              '</div>',unsafe_allow_html=True)
+    
+    st.plotly_chart(fig1_1, theme="streamlit", use_container_width=True)
+
+# Dataset selector
+choice = None
+with row4:
+    st.markdown("<h3 style='text-align: left'>Select one of our datasets and check it's metrics below:</h3>",
+                unsafe_allow_html=True)
+    choice = st.selectbox(
+        "Select one of our datasets",
+        datasets)
+
+selected_ds = choice.split(",")[0].split(":")[1].strip()     
+# Dataset details
+with row4_1.container():    
+  if choice:
+      st.subheader("Dataset: {}".format(selected_ds))
+      st.write(sl.get(selected_ds).manifest)
+with row4_2:
+    
+    if choice:
+        counter = 0
+        random_step = random.randrange(1, 10)
+        txt = ""
+        meta = {}
+
+        ds = sl.get(selected_ds).samples
+        for doc in ds:
+            txt = doc['text'] 
+            meta = doc['meta']
+            counter += 1
+            if counter == random_step:
+                break
+
+        st.subheader("Random document (max 200 chars):")
+        #st.write(ftfy.fix_encoding(txt[:200]))

```

Comment:
To potrzebne do naprawienia wyświetlania sampli. W pewnych przypadkach są znaki które nie są poprawnie rozpoznawane jako utf-8 podczas rysowania.
![image](https://github.com/speakleash/speakleash-dashboard/assets/24353673/3ae877bc-e125-479f-88da-fbcf016aa3b6)

Reply:
Poprawione, wróciłem do poprzedniej wersji tj. `st.write(ftfy.fix_encoding(txt[:200]))`